lib/samples/query_related_features/query_related_features.dart

```diff
@@ -0,0 +1,280 @@
+          spatialReference: SpatialReference.webMercator,
+        ),
+        scale: 36764077,
+      );
+
+    // set selection color
+    _mapViewController.selectionProperties =
+        SelectionProperties(color: Colors.yellow);
+
+    // Set the ready state variable to true to enable the sample UI.
+    setState(() => _ready = true);
+  }
+
+  void onTap(Offset offset) async {
+    // Clear the selection on the feature layer.
+    _alaskaNationalParksLayer.clearSelection();
+
+    // Do an identify on the feature layer and select a feature.
+    final identifyLayerResult = await _mapViewController.identifyLayer(
+      _alaskaNationalParksLayer,
+      screenPoint: offset,
+      tolerance: 12.0,
+      maximumResults: 1,
+    );
+
+    // If there are features identified, show the bottom sheet to display the
+    // attachment information for the selected feature.
+    final features =
+        identifyLayerResult.geoElements.whereType<Feature>().toList();
+    if (features.isNotEmpty) {
+      _alaskaNationalParksLayer.selectFeatures(features: features);
+      final selectedFeature = features.first as ArcGISFeature;
+      setState(() {
+        _layerDataVisible = true;
+        _loadingFeatures = true;
+      });
+      // Query for related features.
+      queryRelatedFeatures(selectedFeature);
+    } else {
+      setState(() {
+        _layerDataVisible = false;
+        _loadingFeatures = false;
+      });
+    }
+  }
+
+  // Query for related features given the origin feature.
+  void queryRelatedFeatures(ArcGISFeature selectedPark) async {
+    // Query for related features.
+    final selectedParkTable = selectedPark.featureTable as ServiceFeatureTable;
+    final relatedFeatureQueryResult =
+        await selectedParkTable.queryRelatedFeatures(feature: selectedPark);
+
+    // Get the related species and preserves features.
+    final lists = <List<String>>[];
+    for (final RelatedFeatureQueryResult result in relatedFeatureQueryResult) {
+      final list = <String>[];
+      for (final feature in result.features()) {
+        ArcGISFeature relatedFeature = feature as ArcGISFeature;
+        // Get a reference to the feature's table.
+        ArcGISFeatureTable relatedTable =
+            feature.featureTable as ArcGISFeatureTable;
+
+        // Get the display field name - this is the name of the field that is intended for display.
+        final displayFieldName = relatedTable.layerInfo!.displayFieldName;
+
+        // Get the display name for the feature.
+        final featureDisplayname = relatedFeature.attributes[displayFieldName];
+
+        // Add the display name to the list.
+        list.add(featureDisplayname);
+      }
+      lists.add(list);
+    }
+
+    // Update the UI with the related features.
+    setState(() {
+      _loadingFeatures = false;
+      _selectedParkName = selectedPark.attributes['UNIT_NAME'];
+
+      _featurePreserves
+        ..clear()
+        ..addAll(lists[0]);
```

Comment:
I think this could just be `_featurePreserves = lists[0];` (and the same for _featureSpecies)